src/metrics.cpp

```diff
@@ -218,8 +218,13 @@ int printStats(bool mining)
 
     if (IsInitialBlockDownload()) {
         int netheight = EstimateNetHeight(height, tipmediantime, Params());
+        int nheaders = mapBlockIndex.size();
```

Comment:
`mapBlockIndex` needs lock on `cs_main`

Reply:
Oops. I've just added the lock. Thanks for the review!